src/Payments.sol

```diff
@@ -234,36 +315,47 @@ contract Payments is
         noRailModificationInProgress(railId)
     {
         Rail storage rail = rails[railId];
+
+        // operator can only reduce fixed lockup on a terminated rail
+        // it cannot change the lockup period or increase fixed lockup
+        require(
+            rail.terminationEpoch == 0 || (period == rail.lockupPeriod && lockupFixed <= rail.lockupFixed),
+            "failed to modify terminated rail: cannot change period or increase fixed lockup"
+        );
+
         Account storage payer = accounts[rail.token][rail.from];
         OperatorApproval storage approval = operatorApprovals[rail.token][
             rail.from
         ][rail.operator];
 
-        require(approval.isApproved, "operator not approved");
-
         // settle account lockup and if account lockup is not settled upto to the current epoch; revert
-        (bool fullySettled, uint256 lockupSettledUpto) = settleAccountLockup(
-            payer
-        );
+        uint256 lockupSettledUpto = settleAccountLockup(payer);
         require(
-            fullySettled && lockupSettledUpto == block.number,
+            lockupSettledUpto == block.number,
             "cannot modify lockup as client does not have enough funds for current account lockup"
         );
 
         // Calculate the change in base lockup
         uint256 oldLockup = rail.lockupFixed +
             (rail.paymentRate * rail.lockupPeriod);
```

Comment:
This old lockup is wrong for terminated rails because the period shrinks over time. Instead of `calculateTerminatedRailLockup`, I think we want a `getEffectiveLockupPeriod(rail)` function that returns the effective lockup period of the rail whether or not it has been terminated. Then we can use that same period for both old lockup and new lockup.

Reply:
Also, we should probably rename these variables to `oldFutureLockup` and `newFutureLockup`. They don't account for lockup that goes towards _existing_ (past) debts.